fiftyone/core/map/mapper.py

```diff
@@ -168,8 +168,18 @@ def map_samples(
             Iterator[Tuple[bson.ObjectId, R]]: The sample ID and the result of
               the map function for the sample.
         """
+        try:
+            result = check_if_return_is_sample(sample_collection, map_fcn)
+        except ValueError as e:
+            if "is empty" in str(e):
```

Comment:
Thank you for catching this, but we really shouldn't be looking for specific strings in errors. I understand we might not want to raise for any generic `ValueError`...since we're close to a release, can we at least make a followup ticket to address this more specifically?